test_bst.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import pytest
+import bst
+
+
+@pytest.fixture(scope="function")
+def initial_node():
+    initial_node = bst.Node(10)
+    return initial_node
+
+
+@pytest.fixture(scope="function")
+def built_tree():
+    tree_root = bst.Node(10)
+    vals = [5, 15, 3, 7, 12, 20, 2, 4, 6, 9, 11, 14, 18, 22]
+    for val in vals:
+        tree_root.insert(val)
+    return tree_root
+
+
+def test_add_same_node(initial_node):
+    initial_node.insert(10)
+    assert initial_node.left is None
+    assert initial_node.right is None
+
+
+def test_add_new_node(initial_node):
+    initial_node.insert(11)
+    assert initial_node.left is None
+    assert initial_node.right.value == 11
+
+
+def test_size(initial_node):
+    assert initial_node.size() == 1
+    initial_node.insert(11)
+    assert initial_node.size() == 2
+
+
+def test_contains(initial_node):
+    assert initial_node.contains(11) is False
+    initial_node.insert(11)
+    assert initial_node.contains(11) is True
+
+
+def test_depth(initial_node):
+    assert initial_node.depth() == 1
+    initial_node.insert(11)
+    assert initial_node.depth() == 2
```

Comment:
It would be instructive to set up tests that validate that the depth stays at 2 for some values that you insert and that it increases to 3 for others.